HRInvoiceApp/HRInvoiceApp/AddCompany.xaml.cs

```diff
@@ -19,27 +20,80 @@ public partial class AddCompany : ContentPage
 
         Department department;
         KvK kvk;
-        Province province;
+        List<Province> provinces;
         User user;
         
-        public AddCompany ()
+        public AddCompany()
 		{
-			InitializeComponent ();
+			InitializeComponent();
             SystemNavigationManager.GetForCurrentView().AppViewBackButtonVisibility = AppViewBackButtonVisibility.Visible;
             db = App.Database.GetInstance();
 
             Task.Run(async () =>
             {
-                company = await db.Table<Company>().FirstOrDefaultAsync();
+                user = await db.Table<User>().FirstOrDefaultAsync();
+                provinces = await db.Table<Province>().ToListAsync();
+                if(provinces.Count == 0)
+                {
+                    // Error handling
+                }
+
+                if(user == null)
+                {
+                    await DisplayAlert("Alert", "Voeg eerst een gebruiker toe.", "OK");
+                    var modalPage = new SettingsPage();
+                    await Navigation.PushModalAsync(modalPage);
+                    return;
+                }
                 company = new Company();
+                kvk = new KvK();
+
+                ProvincePicker.ItemsSource = provinces;
+
             });
+
         }
 
         void saveCompany(object sender, EventArgs e)
         {
+            foreach (var view in addCompanyStackLayout.Children)
+            {
+                if (view is Entry)
+                {
+                    Entry entry = (Entry)view;
+                    if (string.IsNullOrWhiteSpace(entry.Text) && entry.Placeholder.Contains("*"))
+                    {
+                        DisplayAlert("Alert", "Graag alle velden met een * invullen.", "OK");
+                        return;
+                    }
+                }
+            }
+            if (KvKNumber.Text.Count() != 8 || !int.TryParse(KvKNumber.Text, out int result))
```

Comment:
Not sure wat er gebeurt als je een spatie in de zipcode doet. Miss ff voor de zekerheid checken? "1234 AB" ipv "1234AB"